### PR TITLE
BIGTOP-4238: Collect metrics for disk I/O and file descriptors

### PR DESCRIPTION
**Pull request by Cat-Drink**

### CODE REVIEW

File: bigtop-manager-server/src/main/java/org/apache/bigtop/manager/server/controller/MonitoringController.java [hunk 1]
```diff
@@ -42,6 +42,21 @@ public class MonitoringController {
     @Operation(summary = "agent healthy", description = "agent healthy check")
     @GetMapping("agenthealthy")
     public ResponseEntity<JsonNode> agentHostsHealthyStatus() {
+        // 以json格式返回   响应结果 参数即为数据  数据格式为 json
```
Comment (wuchunfu) [inline bigtop-manager-server/src/main/java/org/apache/bigtop/manager/server/controller/MonitoringController.java:45]: @Cat-Drink Please change the comment information to English, thanks

File: bigtop-manager-server/src/main/java/org/apache/bigtop/manager/server/controller/MonitoringController.java [hunk 2]
```diff
@@ -42,6 +42,21 @@ public class MonitoringController {
...
         return ResponseEntity.success(monitoringService.queryAgentsHealthyStatus());
     }
+
+    @Operation(summary = "agent Info", description = "agent info query")
+    @GetMapping("agentinfo")
+    public ResponseEntity<JsonNode> queryAgentsInfo() {
+        // 以json格式返回   响应结果 参数即为数据  数据格式为 json
```
Comment (wuchunfu) [inline bigtop-manager-server/src/main/java/org/apache/bigtop/manager/server/controller/MonitoringController.java:52]: Same as above

File: bigtop-manager-server/src/main/resources/application.yml [hunk 3]
```diff
@@ -30,10 +30,10 @@ spring:
...
     banner-mode: log
   datasource:
-    driver-class-name: org.postgresql.Driver
-    url: jdbc:postgresql://localhost:5432/bigtop_manager
-    username: postgres
-    password: postgres
+    driver-class-name: com.mysql.cj.jdbc.Driver
+    url: jdbc:mysql://localhost:3306/bigtop_manager
+    username: root
+    password: root
```
Comment (wuchunfu) [inline bigtop-manager-server/src/main/resources/application.yml:36]: @Cat-Drink Please withdraw the submission of this code, there is no need to modify it here

File: bigtop-manager-server/src/main/java/org/apache/bigtop/manager/server/proxy/PrometheusProxy.java [hunk 4]
```diff
@@ -81,4 +82,262 @@ public JsonNode queryAgentsHealthyStatus() {
...
+                }
+                return objectMapper.createObjectNode().set("iPv4addr",iPv4addrArray);// IPV4地址
+            }
+        }
+        return objectMapper.createObjectNode();
+    }
+    public JsonNode queryAgentsInfo() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        ArrayNode agentsInfo = objectMapper.createArrayNode();
+        JsonNode agents = queryAgents().get("iPv4addr"); // 获取全主机
```
Comment (wuchunfu) [inline bigtop-manager-server/src/main/java/org/apache/bigtop/manager/server/proxy/PrometheusProxy.java:107]: @Cat-Drink Please change the comment information to English, thanks

File: bigtop-manager-server/src/main/java/org/apache/bigtop/manager/server/proxy/PrometheusProxy.java [hunk 5]
```diff
@@ -81,4 +82,262 @@ public JsonNode queryAgentsHealthyStatus() {
...
+                agentInfo.put("physical_cores", agentCpuMetric.get("physical_cores").asText());
+                agentInfo.put("fileOpenDescriptor",agentCpuMetric.get("fileOpenDescriptor").asLong());
+                agentInfo.put("fileTotalDescriptor",agentCpuMetric.get("fileTotalDescriptor").asLong());
+                LocalDateTime instant = Instant.ofEpochSecond(agentCpuValue.get(0).asLong())
+                        .atZone(ZoneId.systemDefault())
+                        .toLocalDateTime();
+                agentInfo.put("time", instant.toString());
+                for (JsonNode agent : agentCpus) {
+                    agentInfo.put(agent.get("metric").get("cpuUsage").asText(), agent.get("value").get(1).asDouble()); // cpu 指标值
+                }
```
Comment (wuchunfu) [inline bigtop-manager-server/src/main/java/org/apache/bigtop/manager/server/proxy/PrometheusProxy.java:241]: @Cat-Drink Please change the comment information to English, thanks

**Review by wuchunfu — APPROVED** [review]
LGTM, @Cat-Drink thanks

**Comment by lhpqaq** [comment]
You may need to run `mvn spotless:apply`.

**Comment by kevinw66** [comment]
The title should start with `BIGTOP-4238`

**Review by wuchunfu — APPROVED** [review]
LGTM

**Comment by fuxiaofengfu** [comment]
on API "/api/monitoring/agentinfo": disk info of AgentInfo could show with separate file system ?
{
      "hostname": "linux-follower-2",
      "iPv4addr": "192.168.65.6",
      "cpuInfo": "QEMU Virtual CPU version 2.5+",
      "time": "2024-10-19T19:18:49",
      "os": "Linux",
      "architecture": "amd64",
      "physical_cores": "1",
      "memIdle": 1500139520,
      "memTotal": 3840593920,
      "diskFreeSpace": 125490970624,
      "diskTotalSpace": 131423035392
    }

DISK IO also could show with file system related ? not only show physical disk. as maybe we don't know which file system used higher Disk IO.

**Comment by Cat-Drink** [comment]
> on API "/api/monitoring/agentinfo": disk info of AgentInfo could display with separate file system ? from { "hostname": "linux-follower-2", "iPv4addr": "192.168.65.6", "cpuInfo": "QEMU Virtual CPU version 2.5+", "time": "2024-10-19T19:18:49", "os": "Linux", "architecture": "amd64", "physical_cores": "1", "memIdle": 1500139520, "memTotal": 3840593920, "diskFreeSpace": 125490970624, "diskTotalSpace": 131423035392 } to { "hostname": "linux-follower-2", "iPv4addr": "192.168.65.6", "cpuInfo": "QEMU Virtual CPU version 2.5+", "time": "2024-10-19T19:18:49", "os": "Linux", "architecture": "amd64", "physical_cores": "1", "memIdle": 1500139520, "memTotal": 3840593920, "device0Name": "/dev/mapper/rl-home" "device0FreeSpace": 125490970624, "device0TotalSpace": 131423035392, "device1Name": "/dev/mapper/rl-root" "device1FreeSpace": 125490970624, "device1TotalSpace": 131423035392, }
> 
> DISK IO could also displayed with file system related ? not only displayed physical disk. as maybe we don't know which file system used higher Disk IO. such as exists disk sda, two part in it. one is /dev/sda1 mount on / , another is /dev/sda2 mount on /home, then we can clear / and /home are using higher disk IO

ok，i understand

**Comment by Cat-Drink** [comment]
> on API "/api/monitoring/agentinfo": disk info of AgentInfo could display with separate file system ? from { "hostname": "linux-follower-2", "iPv4addr": "192.168.65.6", "cpuInfo": "QEMU Virtual CPU version 2.5+", "time": "2024-10-19T19:18:49", "os": "Linux", "architecture": "amd64", "physical_cores": "1", "memIdle": 1500139520, "memTotal": 3840593920, "diskFreeSpace": 125490970624, "diskTotalSpace": 131423035392 } to { "hostname": "linux-follower-2", "iPv4addr": "192.168.65.6", "cpuInfo": "QEMU Virtual CPU version 2.5+", "time": "2024-10-19T19:18:49", "os": "Linux", "architecture": "amd64", "physical_cores": "1", "memIdle": 1500139520, "memTotal": 3840593920, "device0Name": "/dev/mapper/rl-home" "device0FreeSpace": 125490970624, "device0TotalSpace": 131423035392, "device1Name": "/dev/mapper/rl-root" "device1FreeSpace": 125490970624, "device1TotalSpace": 131423035392, }
> 
> DISK IO could also displayed with file system related ? not only displayed physical disk. as maybe we don't know which file system used higher Disk IO. such as exists disk sda, two part in it. one is /dev/sda1 mount on / , another is /dev/sda2 mount on /home, then we can clear / and /home are using higher disk IO

Regarding Disk IO, we can obtain information about devices such as /dev/sda, /dev/sdb, etc.

**Review by kevinw66 — APPROVED** [review]

**Comment by kevinw66** [comment]
Thanks @Cat-Drink @lhpqaq @fuxiaofengfu @wuchunfu

**Pull request merged**